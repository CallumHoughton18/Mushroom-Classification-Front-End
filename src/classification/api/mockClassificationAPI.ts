import {FormContents} from "../../shared/types";
import {IClassificationAPI, IFeatureDefinition} from "../interfaces";

const mockClassificationAPI: IClassificationAPI = {
    GetClassification: async (data: FormContents): Promise<boolean> => {
        console.log(data);
        await new Promise((resolve) => setTimeout(resolve, 400));
        return true;
    },
    GetClassificationFormDefinition: async (): Promise<IFeatureDefinition[]> => {
        const mockDefs: IFeatureDefinition[] = [];
        for (let index = 0; index < 3; index++) {
            const classificationQuestions: IFeatureDefinition = {
                name: `classques-${index}`,
                options: []
            };
            mockDefs[index] = classificationQuestions;
        }
        return mockDefs;
    }
};

export default mockClassificationAPI;
