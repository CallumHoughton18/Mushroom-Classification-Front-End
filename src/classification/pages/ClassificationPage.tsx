import React, {FunctionComponent} from "react";
import ClassificationForm from "../components/ClassificationForm";
import useAppNavigation from "../../navigation/hooks/useAppNavigation";
import {FormContents} from "../../shared/types";
import {ClassificationQueryData} from "../types";
import {IClassificationAPI} from "../interfaces";
import {useGetFormDefinition} from "../api/classificationAPIHooks";
import {LoadingState} from "../../shared/enums";
import Spinner from "../../shared/components/UI/Spinner";
import useMoreInfoModal from "../hooks/useMoreInfoModal";
import getClassificationDescriptions from "../helpers/getClassificationDescriptions";

type ClassificationPageProps = {
    classificationAPI: IClassificationAPI;
};

const ClassificationPage: FunctionComponent<ClassificationPageProps> = ({classificationAPI}) => {
    const classificationDescs = getClassificationDescriptions();

    const navManager = useAppNavigation();
    const [modal, setQuesForInfo, toggle] = useMoreInfoModal(classificationDescs);
    const [formQuestions, loading] = useGetFormDefinition(classificationAPI, () => {
        navManager.goToErrorPage({message: "Error fetching form definition"});
    });

    if (loading === LoadingState.LOADING) return <Spinner />;

    const navToClassificationResult = (formData: FormContents) => {
        const classificationData: ClassificationQueryData = {
            classificationData: formData
        };

        navManager.goToClassificationResultPage(classificationData);
    };

    return (
        loading === LoadingState.LOADED && (
            <div>
                {modal}
                <ClassificationForm
                    questions={formQuestions}
                    onSubmit={navToClassificationResult}
                    hasInfoPopup={(question) => {
                        return classificationDescs[question.fieldName] != null;
                    }}
                    viewInfoCallback={(question) => {
                        console.log(question);
                        setQuesForInfo(question);
                        toggle();
                    }}
                ></ClassificationForm>
            </div>
        )
    );
};

export default ClassificationPage;
