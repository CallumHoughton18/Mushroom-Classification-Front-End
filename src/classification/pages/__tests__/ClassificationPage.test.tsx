import {render, RenderResult, act, waitForElement, waitFor} from "@testing-library/react";
import React from "react";
import ClassificationPage from "../ClassificationPage";
import mockClassificationAPI from "../../../classification/api/mockClassificationAPI";

jest.mock("../../../navigation/hooks/useAppNavigation");

describe("<ClassificationPage /> render tests", () => {
    it("Should render ClassificationPage component", () => {
        const result = render(<ClassificationPage classificationAPI={mockClassificationAPI} />);
        waitFor(() => expect(result).toMatchSnapshot());
    });
});
