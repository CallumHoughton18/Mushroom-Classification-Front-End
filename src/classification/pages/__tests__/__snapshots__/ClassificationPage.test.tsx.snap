// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClassificationPage /> render tests Should render ClassificationPage component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form>
          <input
            type="submit"
            value="Submit"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form>
        <input
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ClassificationPage /> render tests Should render ClassificationPage component 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form>
          <label
            for="classques-0"
          >
            Classques 0
             
            <i
              class="fa"
            />
          </label>
          <select
            id="classques-0"
            name="classques-0"
            required=""
            title="Classques 0"
          >
            <option
              disabled=""
              value=""
            >
              Select an option
            </option>
          </select>
          <label
            for="classques-1"
          >
            Classques 1
             
          </label>
          <select
            id="classques-1"
            name="classques-1"
            required=""
            title="Classques 1"
          >
            <option
              disabled=""
              value=""
            >
              Select an option
            </option>
          </select>
          <label
            for="classques-2"
          >
            Classques 2
             
            <i
              class="fa"
            />
          </label>
          <select
            id="classques-2"
            name="classques-2"
            required=""
            title="Classques 2"
          >
            <option
              disabled=""
              value=""
            >
              Select an option
            </option>
          </select>
          <input
            type="submit"
            value="Submit"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form>
        <label
          for="classques-0"
        >
          Classques 0
           
          <i
            class="fa"
          />
        </label>
        <select
          id="classques-0"
          name="classques-0"
          required=""
          title="Classques 0"
        >
          <option
            disabled=""
            value=""
          >
            Select an option
          </option>
        </select>
        <label
          for="classques-1"
        >
          Classques 1
           
        </label>
        <select
          id="classques-1"
          name="classques-1"
          required=""
          title="Classques 1"
        >
          <option
            disabled=""
            value=""
          >
            Select an option
          </option>
        </select>
        <label
          for="classques-2"
        >
          Classques 2
           
          <i
            class="fa"
          />
        </label>
        <select
          id="classques-2"
          name="classques-2"
          required=""
          title="Classques 2"
        >
          <option
            disabled=""
            value=""
          >
            Select an option
          </option>
        </select>
        <input
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ClassificationPage /> render tests Should render ClassificationPage component 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body />,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
